basculehttp/bearerTokenFactory_test.go

```diff
@@ -234,3 +235,73 @@ good:
 		})
 	}
 }
+
+func TestProvideResolver(t *testing.T) {
+	type In struct {
+		fx.In
+		R clortho.Resolver `name:"key_resolver"`
+	}
+
+	const yaml = `
+good:
+  factory:
+    uri: "http://test:1111/keys/{keyId}"
+  purpose: 0
+  updateInterval: 604800000000000
+`
+	v := viper.New()
+	v.SetConfigType("yaml")
+	require.NoError(t, v.ReadConfig(strings.NewReader(yaml)))
+
+	goodResolver, err := clortho.NewResolver(clortho.WithKeyIDTemplate("good"))
+	require.Nil(t, err)
+
+	tests := []struct {
+		description      string
+		key              string
+		optional         bool
+		expectedResolver clortho.Resolver
+		expectedErr      error
+	}{
+		{
+			description:      "Success",
+			key:              "good",
+			optional:         false,
+			expectedResolver: goodResolver,
+		},
+		{
+			description: "Silent failure",
+			key:         "bad",
+			optional:    true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			result := In{}
+			assert := assert.New(t)
+			require := require.New(t)
+			app := fx.New(
```

Comment:
you can also use the `fxtest` package, which simplifies a lot of this kind of testing code.  `fxtest.New(...)` does a lot of setup around logging and assertions that will reduce the code here in this test.

Reply:
Thanks for the tip! I'm looking into the `fxtest` package but will leave the test as is for now until I completely understand the `fxtext` package